backends/nats/nats.go

```diff
@@ -0,0 +1,486 @@
+package nats
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"github.com/PowerDNS/simpleblob"
+	"github.com/go-logr/logr"
+	"github.com/nats-io/nats.go"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// DefaultMaxReconnects - Max Reconnect attempts
+	DefaultMaxReconnects = 5
+	// DefaultReconnectWaitSeconds - Time to wait before attempting reconnect
+	DefaultReconnectWaitSeconds = 1
+	// DefaultNatsBucketDescription - Default description for bucket
+	DefaultNatsBucketDescription = "NATS bucket for simpleblob"
+	// DefaultNatsBucketReplicas - Default replicas for bucket
+	DefaultNatsBucketReplicas = 1
+	// DefaultMaxWaitSeconds - Default max wait for Jetstream replies
+	DefaultMaxWaitSeconds = 10
+)
+
+// internal enum to track auth type
+type authType int
+
+const (
+	username authType = iota
+	token
+	nkey
+	credentials
+	nkeyJwt
+	jwt
+	invalid
```

Comment:
Global enums should have a common prefix for readability, otherwise it is hard to understand where that `token` is suddenly coming from, e.g. `AuthUsername`, `AuthToken`, etc.